xds/internal/resolver/xds_resolver.go

```diff
@@ -178,6 +196,13 @@ func (r *xdsResolver) sendNewServiceConfig(cs *configSelector) bool {
 	state := iresolver.SetConfigSelector(resolver.State{
 		ServiceConfig: r.cc.ParseServiceConfig(string(sc)),
 	}, cs)
+
+	// Include the xds client for the LB policies to use.  For unit tests,
+	// r.client may not be a full *xdsclient.Client, but it will always be in
+	// production.
```

Comment:
This comment is very useful, and it took me a little bit of time when reading the cds balancer code (which does not have this comment) to understand what was going on. Maybe we could add this comment everywhere where we do this.

Reply:
> Maybe we could add this comment everywhere where we do this.

Added (to cds).